glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -526,14 +526,33 @@ open class GleanInternalAPI internal constructor () {
      */
     @Suppress("EXPERIMENTAL_API_USAGE")
     internal fun submitPingsByName(pingNames: List<String>) = Dispatchers.API.launch {
+        submitPingsByNameSync(pingNames)
+    }
+
+    /**
+     * Collect and submit a list of pings for eventual upload by name, synchronously.
+     *
+     * Each ping will be looked up in the known instances of [PingType]. If the
+     * ping isn't known, an error is logged and the ping isn't queued for uploading.
+     *
+     * The ping content is assembled as soon as possible, but upload is not
+     * guaranteed to happen immediately, as that depends on the upload
+     * policies.
+     *
+     * If the ping currently contains no content, it will not be assembled and
```

Comment:
This is, as of now, only half correct (less relevant for the code of this PR though).
We can send empty pings now (`send_if_empty`).
I don't know how to properly phrase it though.